Metadata/FormMetadataLoader.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AutomationBundle\Metadata;
+
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FieldMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FormMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\FormMetadataLoaderInterface;
+use Sulu\Bundle\AdminBundle\Metadata\FormMetadata\OptionMetadata;
+use Sulu\Bundle\AdminBundle\Metadata\MetadataInterface;
+use Sulu\Bundle\AutomationBundle\TaskHandler\AutomationTaskHandlerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Task\TaskBundle\Handler\TaskHandlerFactory;
+
+class FormMetadataLoader implements FormMetadataLoaderInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var TaskHandlerFactory
+     */
+    private $taskHandlerFactory;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        TaskHandlerFactory $taskHandlerFactory
+    ) {
+        $this->translator = $translator;
+        $this->taskHandlerFactory = $taskHandlerFactory;
+    }
+
+    public function getMetadata(string $key, string $locale, array $metadataOptions): ?MetadataInterface
+    {
+        if (0 !== strcmp('task_details', $key)) {
+            return null;
+        }
+
+        /** @var FormMetaData $form */
+        $form = new FormMetadata();
+
+        // Single Select
+        $singleSelectHandler = new FieldMetadata('handlerClass');
+        $singleSelectHandler->setType('single_select');
+        $singleSelectHandler->setLabel($this->translator->trans('sulu_automation.task.name', [], 'admin', $locale));
+        $singleSelectHandler->setRequired(true);
+
+        $defaultValueOption = new OptionMetadata();
```

Comment:
this sets the default value of the `single_select` to `''` right? does that change anything? 🙈